File: Algorithms/Medium/128.Longest-Consecutive-Sequence/LongestConsecutiveSequence.py
# Time complexity: O(n)
# Space complexity: O(n)
class Solution:
    def longestConsecutive(self, nums: list[int]) -> int:
        hash_set = set(nums)
        max_length = 0

        for num in hash_set:
            if (num - 1) not in hash_set:
                next_num = num + 1
                while next_num in hash_set:
                    next_num += 1

                max_length = max(max_length, next_num - num)

        return max_length

File: Algorithms/Medium/128.Longest-Consecutive-Sequence/LongestConsecutiveSequence.go
package medium

// Time complexity: O(n)
// Space complexity: O(n)
func longestConsecutive(nums []int) int {
	hashSet := make(map[int]bool)
	longest_seq := 0

	for _, num := range nums {
		if !hashSet[num] {
			hashSet[num] = true
		}
	}

	for _, num := range nums {
		if !hashSet[num-1] {
			next := num + 1
			for hashSet[next] {
				next++
			}

			longest_seq = max(longest_seq, next-num)
		}
	}

	return longest_seq
}
